src/main/java/com/openshift/jenkins/plugins/pipeline/Deployment.java

```diff
@@ -148,6 +149,11 @@ public static boolean didAllImagesChangeIfNeeded(String buildConfig, TaskListene
 			
 			if (didImageChangeFromPreviousVersion(client, dc.getLatestVersionNumber(), 
 					chatty, listener, dc.getName(), namespace, latestImageHexID, imageTag)) {
+
+				if(annotateRC) {
+					IReplicationController rc = getLatestReplicationController(dc, namespace, client, chatty ? listener : null);
+    				AnnotationUtils.AnnotateResource(client, listener, chatty, env, rc); 
```

Comment:
@gabemontero seems like this block might need to move as well to ensure we annotate the RC even if the image didn't change?